src/codewhisperer/util/customizationUtil.ts

```diff
@@ -301,12 +301,25 @@ export const selectCustomization = async (customization: Customization) => {
 export const getAvailableCustomizationsList = async () => {
     const items: Customization[] = []
     const response = await codeWhispererClient.listAvailableCustomizations()
```

Comment:
can the client automatically log all calls, similar to the codecatalyst client? instead of logging at this higher layer in an ad-hoc way.

Reply:
sg, will look into how codecatalyst take care of it

Reply:
in the CC client all requests go through a `call()` abstraction: https://github.com/aws/aws-toolkit-vscode-staging/blob/a717af3883355d8d3bf69fa444e1841b25dc63dd/src/shared/clients/codecatalystClient.ts#L222-L230

I think the SDKs also support basic logging with a flag, so that could be another way, but it's less customizable.

Ideally we would have a base "client" that does this for all clients, but that is TODO.

Reply:
yea I have looked it, but @justinmk3  I think I will need more time to improve our CW client facade to be like CodeCatlyst's. I still moved the logging logic inside CW client facade `codewhisperer.ts` as an interim solution  tho, just the functionality is not as comprehensive as `codecatlystClient.ts` is. Let me cut a backlog item for it

Reply:
will share the SIMs(including the codeScan one) in our channel